Code/MinimalLib/common.h

```diff
@@ -67,6 +67,20 @@
 #pragma GCC diagnostic pop
 #endif
 #endif
+
+#define GET_JSON_VALUE(doc, key, type)                                     \
+  const auto key##It = doc.FindMember(#key);                               \
+  if (key##It != doc.MemberEnd()) {                                        \
+    if (!key##It->value.Is##type()) {                                      \
+      return "JSON contains '" #key "' field, but its type is not '" #type \
+             "'";                                                          \
+    }                                                                      \
+    key = key##It->value.Get##type();                                      \
+  }
+
+#define GET_JSON_VALUE_WITH_DEFAULT(doc, key, type, defaultValue) \
```

Comment:
oh boy... that's pretty gross.
👍

Reply:
Don't pretend you don't like that horrible macro, I would not believe you anyway :)

Reply:
> Don't pretend you don't like that horrible macro, I would not believe you anyway :)

The fact that you nested them was the real 🍒 on the 🍰